File: Booth.DockerVolumeBackup.WebApi/Models/BackupStatus.cs
namespace Booth.DockerVolumeBackup.WebApi.Models
{

    public enum Status { Queued, Active, Complete, Error}

    public class BackupStatus
    {
        public int BackupId { get; set; }
        public Status Status { get; set; }
        public List<VolumeBackupStatus> Volumes { get; set; } = new List<VolumeBackupStatus>();
    }

    public class VolumeBackupStatus
    {
        public string VolumeName { get; set; } = "";
        public Status Status { get; set; }
        public DateTimeOffset? BackupTime { get; set; }
    }
}

File: booth.dockervolumebackup.ui/src/models/BackupStatus.ts
import { Status } from '@/models/Backup';
export interface BackupStatus {
    backupId: number;
    status: Status;
    startTime?: Date;
    endTime?: Date;
    volumes: VolumeBackupStatus[];
}

export interface VolumeBackupStatus {
    backupVolumeId: number;
    status: Status;
    startTime?: Date;
    endTime?: Date;
}